SConstruct

```diff
@@ -37,13 +37,15 @@ def set_defaults(env):
     env.Append(CCFLAGS=['-g', '-Wshadow', '-Wall', '-Wno-missing-braces',
                         '-fpic', '-D_GNU_SOURCE', '-DD_LOG_V2'])
     env.Append(CCFLAGS=['-O2', '-DDAOS_VERSION=\\"' + DAOS_VERSION + '\\"'])
+    env.Append(CCFLAGS=['-DCMOCKA_FILTER_SUPPORTED=0'])
```

Comment:
If cmocka version >= 1.1.5 then cmocka filters are supported, however I believe the version that comes with most distro repos is less than this. This define is to include/exclude the cmocka functionality and is excluded by default.
My initial approach was to not include the compile time flag at all and only add it locally when I wanted to use the filters, but I kept forgetting what the define was and had to look it up each time I wanted to use it. This approach is easier for me to turn it on, but if it's not appropriate here it can be removed.